### ISSUE TITLE
Gini Importance / MDI

### ISSUE THREAD

**Issue by nalzok** [dmlc/xgboost#5017]
How can I get the Gini Importance, a.k.a. MDI, from a booster model?

According to the [document](https://xgboost.readthedocs.io/en/latest/tutorials/model.html#learn-the-tree-structure), it appears that the gain of a leaf is simply MDI plus gamma, so I'm tempted to use `xgb.importance(model = bst)$Gain + bst$params$gamma`. However, that could be inaccurate since `xgb.importance` calculates the relative gain which sum to 1, but would that work for the scenario where gamma is zero?

**Comment by hcho3** [comment]
XGBoost does not use Gini importance to obtain optimal splits. Instead, it uses first and second-order gradient of the loss function to evaluate splits.
